aviary/subsystems/energy/battery_builder.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import openmdao.api as om
+
+from aviary.subsystems.subsystem_builder_base import SubsystemBuilderBase
+from aviary.subsystems.energy.battery_sizing import SizeBattery
+from aviary.variable_info.variables import Aircraft, Dynamic
+
+
+class BatteryBuilder(SubsystemBuilderBase):
+    def build_pre_mission(self, aviary_inputs=None):
+        return SizeBattery(aviary_inputs=aviary_inputs)
+
+    def get_mass_names(self):
+        return [Aircraft.Battery.MASS]
+
+    def build_mission(self, num_nodes, aviary_inputs=None) -> om.Group:
+        battery_group = om.Group()
+        soc = om.ExecComp('state_of_charge = (energy_capacity - (electric_energy/efficiency)) / energy_capacity',
+                          state_of_charge={'val': np.zeros(
```

Comment:
is the storage efficiency, charge efficiency, discharge efficiency, etc?

Reply:
When originally making this model I didn't go into that level of detail and just added in a universal efficiency factor so at least there is something for users to play with. I'd argue if you need that level of detail you should start looking into higher-fidelity battery models such as the external subsystem one

Reply:
maybe just add a comment here then, something like "here efficiency is used an an average (or overall) efficiency"

Reply:
Done with "# Here, the efficiency variable is used as an overall efficiency for the battery"